testsuite/cluster-test/src/load_test/mod.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::future::try_join_all;
+use std::fmt::Display;
+use std::time::Duration;
+
+mod mempool;
+mod state_sync;
+mod tx_emitter;
+
+#[async_trait]
+pub trait Handler: Display + Send {
+    async fn start(&mut self, duration: Duration) -> Result<()>;
```

Comment:
the interface seems too general, what I had in mind is just
```
enum {
  JSON_RPC,
  LibraNet,
}
trait LoadTest: {
  type Load;
  fn endpoint() -> Endpoint; 
  fn next_load() -> Self::Load;
}
```
and implement the general handling like
```
fn run(load_test: Box<dyn LoadTest>) {
  while let Some(load) = load_test.generate_load() {
    endpoint.submit(load);
  }
}
```

Reply:
I like the enum and load test type, but my thinking is for each type, we leave a handler and CT only needs care about handler start and end within an input duration.

for example user can pass in multi flags stand for each component which they wanna include in load test. And test param will build up a vec of handlers. We give flexibility  of each component load generation part to user since CT side does not have full sight of how does each component actually generate load, do they have common points to abstract. So this is why I called it version 1, which it looks a little general, but we can extend or refactory easier